File: tools/testy/sample/SimpleTest.cpp
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */


#include "nspr.h"
#include "TestySupport.h"

PR_BEGIN_EXTERN_C
PR_EXPORT(int) Testy_Init();
PR_EXPORT(int) Testy_RunTest();
PR_EXPORT(int) Testy_Shutdown();
PR_END_EXTERN_C


int Testy_Init() 
{
  return 0;
}


int Testy_RunTest() 
{
  const char* testCaseName1 = "Simple Test 1";

  Testy_LogStart(testCaseName1);
  Testy_LogComment(testCaseName1, "Comment 1");
  Testy_LogEnd(testCaseName1, true);

  const char* testCaseName2 = "Simple Test 2";

  Testy_LogStart(testCaseName2);
  Testy_LogComment(testCaseName2, "Comment 2");
  Testy_LogEnd(testCaseName2, false);

  return 0;
}

int Testy_Shutdown() 
{
  return 0;
}

File: tools/lint/eslint/eslint-plugin-mozilla/lib/environments/simpletest.js
/**
 * @fileoverview Defines the environment for scripts that use the SimpleTest
 *               mochitest harness. Imports the globals from the relevant files.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

"use strict";

// -----------------------------------------------------------------------------
// Rule Definition
// -----------------------------------------------------------------------------

var path = require("path");
var { getScriptGlobals } = require("./utils");

// When updating this list, be sure to also update the 'support-files' config
// in `tools/lint/eslint.yml`.
const simpleTestFiles = [
  "AccessibilityUtils.js",
  "ExtensionTestUtils.js",
  "EventUtils.js",
  "MockObjects.js",
  "SimpleTest.js",
  "WindowSnapshot.js",
  "paint_listener.js",
];
const simpleTestPath = "testing/mochitest/tests/SimpleTest";

module.exports = getScriptGlobals(
  "simpletest",
  simpleTestFiles.map(file => path.join(simpleTestPath, file))
);
